manifests/extension.pp

```diff
@@ -95,9 +95,11 @@
       $real_package = "${package_prefix}${title}"
     }
 
-    ensure_resource('package', $header_packages)
-    Package[$header_packages] -> Package[$real_package] -> ::Php::Config[$title]
-
+    unless empty($header_packages) {
+      ensure_resource('package', $header_packages)
+      Package[$header_packages] -> Package[$real_package]
```

Comment:
if header_packages are empty, the dependency Package[$real_package] -> ::Php::Config[$title] is not defined.